app/src/main/java/com/github/moagrius/tileview3/ScrollView.java

```diff
@@ -0,0 +1,550 @@
+
+  private static final int DIRECTION_BACKWARD = -1;
+  private static final int DIRECTION_FORWARD = 1;
+
+  private boolean mIsFlinging;
+  private boolean mIsDragging;
+
+  private boolean mSmoothScrollingEnabled = true;
+
+  private long mLastScrolledAt;
+
+  private OverScroller mOverScroller;
+
+  private GestureDetector mGestureDetector;
+  private TouchUpGestureDetector mTouchUpGestureDetector;
+
+  /**
+   * Constructor to use when creating a ScrollView from code.
+   *
+   * @param context The Context the ScrollView is running in, through which it can access the current theme, resources, etc.
+   */
+  public ScrollView(Context context) {
+    this(context, null);
+  }
+
+  public ScrollView(Context context, AttributeSet attrs) {
+    this(context, attrs, 0);
+  }
+
+  public ScrollView(Context context, AttributeSet attrs, int defStyleAttr) {
+    super(context, attrs, defStyleAttr);
+    setWillNotDraw(false);
+    setClipChildren(false);
+    setFocusable(true);
+    setDescendantFocusability(FOCUS_AFTER_DESCENDANTS);
+    mGestureDetector = new GestureDetector(context, this);
+    mTouchUpGestureDetector = new TouchUpGestureDetector(this);
+    mOverScroller = new OverScroller(context);
+  }
+
+  private void assertSingleChild() {
+    if (getChildCount() > 0) {
+      throw new IllegalStateException(ADD_VIEW_ERROR_MESSAGE);
+    }
+  }
+
+  @Override
+  public void addView(View child) {
+    assertSingleChild();
+    super.addView(child);
+  }
+
+  @Override
+  public void addView(View child, int index) {
+    assertSingleChild();
+    super.addView(child, index);
+  }
+
+  @Override
+  public void addView(View child, ViewGroup.LayoutParams params) {
+    assertSingleChild();
+    super.addView(child, params);
+  }
+
+  @Override
+  public void addView(View child, int index, ViewGroup.LayoutParams params) {
+    assertSingleChild();
+    super.addView(child, index, params);
+  }
+
+  public void setOverScroller(OverScroller overScroller) {
+    mOverScroller = overScroller;
+  }
+
+  public boolean isSmoothScrollingEnabled() {
+    return mSmoothScrollingEnabled;
+  }
+
+  public void setSmoothScrollingEnabled(boolean smoothScrollingEnabled) {
+    mSmoothScrollingEnabled = smoothScrollingEnabled;
+  }
+
+  /**
+   * Ask one of the children of this view to measure itself, taking into
+   * account both the MeasureSpec requirements for this view and its padding.
+   * The heavy lifting is done in getChildMeasureSpec.
+   *
+   * @param child                   The child to measure
+   * @param parentWidthMeasureSpec  The width requirements for this view
+   * @param parentHeightMeasureSpec The height requirements for this view
+   */
+  protected void measureChild(View child, int parentWidthMeasureSpec, int parentHeightMeasureSpec) {
+    LayoutParams lp = (LayoutParams) child.getLayoutParams();
+    int childWidthMeasureSpec = getScrollViewChildMeasureSpec(parentWidthMeasureSpec, getPaddingLeft() + getPaddingRight(), lp.width);
```

Comment:
Can't, their package-private.  The getters will be inlined most of the time anyway